lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -287,6 +287,12 @@ def name_from_address(options)
         return address[:name] if address[:name]
       end
 
+      def name_from_payment_method(payment_method)
+        return unless payment_method.first_name && payment_method.last_name
```

Comment:
Do payment methods have a `full_name` attribute in AM or is it just in core?

Reply:
We have `name` which should work instead of needing this method; I'll take this from here.

Reply:
Ah I see now why this is being done; `.name` will return an empty string, which fouls up the conditional. I think this is fine.